desktop/src/app/app.component.ts

```diff
@@ -68,17 +69,20 @@ export class AppComponent implements OnInit, OnDestroy {
         private ncjTemplateService: NcjTemplateService,
         private predefinedFormulaService: PredefinedFormulaService,
         private workspaceService: WorkspaceService,
+        private appLocaleService: AppLocaleService
     ) {
-        // Initialize shared component lib environment
-        initEnvironment(new DefaultBrowserEnvironment(
+        const localizer = new DesktopLocalizer(this.appLocaleService);
+        localizer.loadTranslations().then(() => {
```

Comment:
This looks like a race condition between isAppready and localization strings being loaded. Maybe use an Observable and put it into the combineLatest() list below?

Reply:
See haveWorkspacesLoaded in WorkSpaceService for an example

Reply:
On second thought, if we can get rid of loadTranslations (see my other comment) then this doesn't need to involve async initialization at all.